albumentations/core/transforms_interface.py

```diff
@@ -85,6 +85,34 @@ def __init__(self, p: float = 0.5, always_apply: bool | None = None):
         self._key2func = {}
         self._set_keys()
         self.processors: dict[str, BboxProcessor | KeypointsProcessor] = {}
+        self.seed: int | None = None
+        self.random_generator = np.random.default_rng(self.seed)
+        self.py_random = random.Random(self.seed)  # Create instance instead of using global
+
+    def set_random_state(self, seed: int | None) -> None:
+        """Set random state for this transform and all nested transforms.
+
+        Args:
+            seed: Random seed to use
+        """
+        if seed is not None:
+            random.seed(seed)  # Set standard library random seed
```

Comment:
It looks like this implementation impacting the global random state? Is that necessary? I would have expected that calling the `set_random_state` on an object would only impact the object's internal state, and not the global state.

Reply:
Cleaned this up in a later Pull request